playground/src/import-url/import-url.js

```diff
@@ -0,0 +1,126 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import './import-url.scss';
+
+import events from '../events/events.js';
+import * as Button from '../button/button.js';
+import * as Document from '../document/document.js';
+import FlyIn from '../fly-in/base.js';
+
+export const EVENT_REQUEST_URL_CONTENT = 'event-request-url-content';
+export const EVENT_UPDATE_EDITOR_CONTENT = 'event-update-editor-content';
+
+/* eslint-disable max-len */
+const URL_VALIDATION_REGEX = /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/gm;
+
+export function createImportURLView(target, trigger) {
+  return new ImportURL(target, trigger);
+}
+
+class ImportURL extends FlyIn {
+  constructor(target, trigger) {
+    super(target);
+
+    this.target = target;
+    this.trigger = Button.from(trigger, this.toggle.bind(this));
+
+    const content = document.createElement('div');
+    content.className = 'import-url';
+    content.innerHTML = `
+      <p class="experimental-view-intro">
+        Enter a valid URL to import the page's markup into the editor.
+      </p>
+      <div class="import-url-bar">
+        <input id="url-bar-input"
+            class="import-url-bar-input"
+            type="url"
+            name="import-url"
+            placeholder="Your URL">
+        <button id="url-bar-submit"
+            class="import-url-bar-submit"
+            type=""
+            name="button">
+          Import
+        </button>
+      </div>
+      <label id="url-bar-label"
+          class="import-url-bar-label"
+          for="url-bar-input">fhwe
+      </label>`;
+
+    this.upadateContent(content);
+
+    this.urlBarLabel = document.getElementById('url-bar-label');
+    this.urlBarInput = document.getElementById('url-bar-input');
+    this.urlBarSubmit = document.getElementById('url-bar-submit');
+
+    this.urlBarSubmit.addEventListener(
+      'click',
+      this.importEventHandler.bind(this)
+    );
+    this.urlBarInput.addEventListener('keyup', (e) => {
+      if (e.keyCode === 13) {
+        this.importEventHandler(e);
+      }
+    });
+
+    events.subscribe(Document.EVENT_RECEIVED_URL_CONTENT, (url, content) => {
+      window.requestIdleCallback(() => {
+        this.receiveContent(url, content);
+      });
+    });
+  }
+
+  importEventHandler(e) {
+    e.preventDefault();
+    const input = this.urlBarInput.value;
+    const url = input.startsWith('http://') || input.startsWith('https://') ? input : `http://${input}`;
+    if (url.match(URL_VALIDATION_REGEX)) {
+      this.importURL(url);
+    } else {
+      this.importError('Please enter a valid URL');
+      this.urlBarInput.focus();
+    }
+  }
+
+  importURL(url) {
+    events.publish(EVENT_REQUEST_URL_CONTENT, url);
+    this.urlBarSubmit.classList.add('loading');
+    this.urlBarSubmit.innerText = '';
```

Comment:
What about using `font-size: 0;` instead on the `.loading` selector? This would reduce the amount of string literals in the application code.